src/DynamoCore/Engine/AstBuilder.cs

```diff
@@ -267,7 +267,14 @@ private void CompileToAstNodes(NodeModel node, List<AssociativeNode> resultList,
         {
 
             var inputAstNodes = new List<AssociativeNode>();
-            foreach (int index in Enumerable.Range(0, node.InPorts.Count))
+            var inPortsCount = node.InPorts.Count;
+            var inPortDataCount = node.InPortData.Count;
+
+            //TODO: inputsCount should be removed in future. 
+            // InPortData won't be used anymore, so we shouldn't take into account InPortData.Count.
+            int inputsCount = inPortsCount > inPortDataCount ? inPortsCount : inPortDataCount;
```

Comment:
For now we can't use just number of ports, because some nodes still use `InPortData` and Color Range use `InPorts`.